cache-config/testing/ort-tests/t3c-reload_test.go

```diff
@@ -0,0 +1,96 @@
+	"strings"
+	"testing"
+
+	"github.com/apache/trafficcontrol/cache-config/t3cutil"
+	"github.com/apache/trafficcontrol/cache-config/testing/ort-tests/tcdata"
+)
+
+func TestT3cReload(t *testing.T) {
+	t.Logf("------------- Starting TestT3cReload ---------------")
+	tcd.WithObjs(t, []tcdata.TCObj{
+		tcdata.CDNs, tcdata.Types, tcdata.Tenants, tcdata.Parameters,
+		tcdata.Profiles, tcdata.ProfileParameters, tcdata.Statuses,
+		tcdata.Divisions, tcdata.Regions, tcdata.PhysLocations,
+		tcdata.CacheGroups, tcdata.Servers, tcdata.Topologies,
+		tcdata.DeliveryServices}, func() {
+		t.Logf("------------- Start TestT3cReload Core ---------------")
+
+		cacheHostName := "atlanta-edge-03"
+
+		t.Logf("DEBUG TestT3cReload calling badass")
+		if stdOut, exitCode := t3cUpdateReload(cacheHostName, "badass"); exitCode != 0 {
+			t.Fatalf("ERROR: t3c badass failed: code '%v' output '%v'\n", exitCode, stdOut)
+		}
+
+		t.Logf("DEBUG TestT3cReload deleting file")
+
+		// delete a file that we know should trigger a reload.
+		fileNameToRemove := filepath.Join(test_config_dir, "hdr_rw_first_ds-top.config")
+		if err := os.Remove(fileNameToRemove); err != nil {
+			t.Fatalf("failed to remove file '" + fileNameToRemove + "': " + err.Error())
+		}
+
+		t.Logf("DEBUG TestT3cReload setting upate flag")
+		// set the update flag, so syncds will run
+		if err := ExecTOUpdater("atlanta-edge-03", false, true); err != nil {
+			t.Fatalf("t3c-update failed: %v\n", err)
+		}
+
+		t.Logf("DEBUG TestT3cReload calling syncds")
+		stdOut, _ := t3cUpdateReload(cacheHostName, "syncds")
+		// Ignore the exit code error for now, because the ORT Integration Test Framework doesn't currently start ATS.
+		// TODO check err, after running ATS is added to the tests.
+		// if err != nil {
+		// 	t.Fatalf("t3c syncds failed: %v\n", err)
+		// }
+
+		t.Logf("DEBUG TestT3cReload looking for reload string")
+		if !strings.Contains(stdOut, `Running 'traffic_ctl config reload' now`) {
+			t.Errorf("expected t3c to do a reload after adding a header rewrite file, actual: '''%v'''\n", stdOut)
+		}
+
+		t.Logf("------------- End TestT3cReload Core ---------------")
+	})
+	t.Logf("------------- End of TestT3cReload ---------------")
+}
+
+func t3cUpdateReload(host string, runMode string) (string, int) {
+	args := []string{
+		"apply",
+		"--traffic-ops-insecure=true",
+		"--dispersion=0",
+		"--login-dispersion=0",
+		"--traffic-ops-timeout-milliseconds=3000",
+		"--traffic-ops-user=" + tcd.Config.TrafficOps.Users.Admin,
+		"--traffic-ops-password=" + tcd.Config.TrafficOps.UserPassword,
+		"--traffic-ops-url=" + tcd.Config.TrafficOps.URL,
+		"--cache-host-name=" + host,
+		"--log-location-error=stdout",
+		"--log-location-info=stdout",
+		"--log-location-debug=test.log",
+		"--omit-via-string-release=true",
+		"--git=" + "yes",
+		"--run-mode=" + runMode,
+	}
+	stdOut, _, exitCode := t3cutil.Do("t3c", args...) // should be no stderr, we told it to log to stdout
+	return string(stdOut), exitCode
+}
```

Comment:
They're not identical, this test needs the exit code and stdout. Using `runApply` will require changing it to return the same, and changing everything that calls it to handle the exit code instead of the Go error.

Reply:
`runApply`also logs to `test.log`, where `t3cUpdateReload` needs to log to stdout for the test. Unifying them would require making everything currently calling `runApply` to open the `test.log` file and write to it manually, or else take the log locations as arguments as well.